src/core/search/compressed_list.cc

```diff
@@ -0,0 +1,210 @@
+#include "core/search/compressed_list.h"
+
+#include <array>
+
+#include "base/logging.h"
+
+namespace dfly::search {
+
+using namespace std;
+
+CompressedList::Iterator::Iterator(const CompressedList& list) : stash_{}, diffs_{list.diffs_} {
```

Comment:
I'd assign `stash_` and `diffs_` a default value (alongside their declaration) and have the default constructor `= default` and here only assign `diffs_`. I think it's safer that way for future changes.